### ISSUE TITLE
Splitting a 'to' member of destination_sign relations adds an extra 'to' member

### ISSUE THREAD

**Issue by fidelis-assis** [openstreetmap/iD#9188]
### How to reproduce the issue?

Just split a way that is a 'to' member of any destination_sign relation.
destination_sign relations can have exactly one 'to' member.
Splitting a 'from' member also adds extra 'from', not sure if that is a problem too.

### Which iD Editor versions do you see the issue on?

Released version at openstreetmap.org/edit

### Which browsers are you seeing this problem on?

Chrome

**Comment by 1ec5** [comment]
At the time of the split, did the relation already have `from`, `via`, and `to` members, or just a `to` member? It looks like the code for splitting only tidies up relations that already have the full complement of members:

https://github.com/openstreetmap/iD/blob/9a26ae3e9ed27373869e48936b7604e20e196c5f/modules/osm/relation.js#L259-L265 https://github.com/openstreetmap/iD/blob/c85456438987715e9d243e519200bb4b9134087e/modules/actions/split.js#L176-L179

[This documentation](https://wiki.openstreetmap.org/wiki/Special:PermanentLink/2314704#Members) points out that it’s valid for there to be no `from` member, which would be unlike any other relation type that uses `from`/`via`/`to` roles. I disagree with the note on that page about “via” nodes being nonsensical – they would make sense for the same reason as “via” nodes on any other kind of relation, such as `restriction` relations. This issue has been raised [on the talk page](https://wiki.openstreetmap.org/wiki/Talk:Relation:destination_sign#'via'_as_a_valid_role_member) as well.

**Comment by fidelis-assis** [comment]
Yes, the problem occurs with complete relations. I've just created a complete one to show the issue, relation 14305516, and then split the to member. The attached print shows the problem.
I also disagree with that note. I think that 'intersection' nodes are important/necessary. 'via' ways are not part of the original proposal, but used "informally". By the way, it'd be nice if iD implemented some validation of these relations.

![extra_to_after_splitting](https://user-images.githubusercontent.com/2800137/176319797-4716957b-feec-459f-8c29-bce1277ffff1.png)

**Comment by 1ec5** [comment]
OK, so the issue is that the role is `intersection` instead of `via`, which causes iD to not recognize the relation as being the sort of relation that has turn restriction–like roles. It’s quite unfortunate that the proposal co-opted `from` and `to` roles but with different rules for the `from` role and a novel `via`-like `intersection` role. We could work around that design problem by equating `via` and `intersection` in the `hasFromViaTo()` method above, but it would be prudent to limit this special case to `destination_sign` relations. I also wonder if we really should place such importance on the relation type’s original design, since the proposal was rejected.

**Comment by fidelis-assis** [comment]
OK, limiting equating  "intersection" and "via" to destinmation_sign relations seems fine. I also agree with not placing much importance to that proposal (rejected but widely used, ex. 31500+ in Germany, 5500+ in USA, 4700+ in France, 1400+ in UK, 850+ in Brazil). Anyway, it'd be interesting to have a minimum validation such as requiring exactly one "to" member.
I personally dislike optional "intersection" and "from" but that's the de facto usage.

**Comment by fidelis-assis** [comment]
Hello, what's the status of this issue? I mean, avoid inclusion of multiples 'from' and 'to' to the destination_sign relation when a way ('from', 'to' or 'via') is splitted, so as not to break/change the relation.
